src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3794,7 +3840,8 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 }
                 tmp = ASR::make_LogicalConstant_t(al, x.base.base.loc, result, type);
                 return;
-            } else if (call_name == "int" || call_name == "float" || call_name == "bool") {
+            } else if ( call_name == "int" || call_name == "float" 
+                        || call_name == "bool" || call_name == "str") {
```

Comment:
String comparison is expensive. 
Let's compare strings only once and assign `tmp` to something.

```c++
} else if (call_name == "int") {
    tmp = handle_intrinsic_int(...);
} else if (call_name == "str") {
    tmp = handle_intrinsic_str(...);
} ... 
```

Reply:
i changed it.